lib/tcpip/network_wrapper.cc

```diff
@@ -460,8 +460,9 @@ Socket network_socket_create_and_bind(Timeout            *timeout,
                                       bool                isListening,
                                       uint16_t            maxConnections)
 {
-	return with_restarting_checks(
-	  [&]() -> Socket {
+	Socket ret = nullptr;
+	with_restarting_checks(
```

Comment:
Is the return of this function lost?

Reply:
Actually we don't care about that return value, the error is already encoded in the socket being null or not.

The function has to return an int to match the signature expected by `with_restarting_checks` to avoid another template instantiation.

I used the differentiation between -EAGAIN, -ENOMEM, etc. for readability, making sense visually of the error paths. But I realize now it may be confusing. Any suggestion?